File: Android/agc-remoteconfig-demo-java/app/src/main/java/com/huawei/agc/quickstart/remoteconfig/RemoteConfigMessageService.java
package com.huawei.agc.quickstart.remoteconfig;

import android.content.SharedPreferences;
import android.util.Log;

import com.huawei.hms.push.HmsMessageService;
import com.huawei.hms.push.HmsMessaging;
import com.huawei.hms.push.RemoteMessage;

import java.util.Arrays;

public class RemoteConfigMessageService extends HmsMessageService {
    private static final String TAG = "MessageService";

    @Override
    public void onMessageReceived(RemoteMessage message) {
        Log.i(TAG, "onMessageReceived is called");
        if (message == null) {
            Log.e(TAG, "Received message entity is null!");
            return;
        }
        Log.i(TAG, "getCollapseKey: " + message.getCollapseKey()
                + "\n getData: " + message.getData()
                + "\n getFrom: " + message.getFrom()
                + "\n getTo: " + message.getTo()
                + "\n getMessageId: " + message.getMessageId()
                + "\n getOriginalUrgency: " + message.getOriginalUrgency()
                + "\n getUrgency: " + message.getUrgency()
                + "\n getSendTime: " + message.getSentTime()
                + "\n getMessageType: " + message.getMessageType()
                + "\n getTtl: " + message.getTtl());
        RemoteMessage.Notification notification = message.getNotification();
        if (notification != null) {
            Log.i(TAG, "\n getImageUrl: " + notification.getImageUrl()
                    + "\n getTitle: " + notification.getTitle()
                    + "\n getTitleLocalizationKey: " + notification.getTitleLocalizationKey()
                    + "\n getTitleLocalizationArgs: " + Arrays.toString(notification.getTitleLocalizationArgs())
                    + "\n getBody: " + notification.getBody()
                    + "\n getBodyLocalizationKey: " + notification.getBodyLocalizationKey()
                    + "\n getBodyLocalizationArgs: " + Arrays.toString(notification.getBodyLocalizationArgs())
                    + "\n getIcon: " + notification.getIcon()
                    + "\n getSound: " + notification.getSound()
                    + "\n getTag: " + notification.getTag()
                    + "\n getColor: " + notification.getColor()
                    + "\n getClickAction: " + notification.getClickAction()
                    + "\n getChannelId: " + notification.getChannelId()
                    + "\n getLink: " + notification.getLink()
                    + "\n getNotifyId: " + notification.getNotifyId());
        }

        if (message.getDataOfMap().containsKey("DATA_STATE")) {
            SharedPreferences preferences = this.getApplicationContext().getSharedPreferences("Remote_Config", MODE_PRIVATE);
            preferences.edit().putBoolean("DATA_OLD", true).apply();
        }
    }

    @Override
    public void onNewToken(String s) {
        super.onNewToken(s);
        Log.d(TAG, "Push Token : " + s);
        HmsMessaging.getInstance(this).subscribe("PUSH_REMOTE_CONFIG").addOnCompleteListener(task -> {
            if (task.isSuccessful()) {
                Log.i(TAG, "subscribe Complete");
            } else {
                Log.e(TAG, "subscribe failed: cause=" + task.getException().getMessage());
            }
        });
    }
}

File: Xamarin/agc-remoteconfig-xamarin/android/XamarinHmsRemoteConfig/RemoteConfigMessageService.cs
/*
       Copyright 2020-2021. Huawei Technologies Co., Ltd. All rights reserved.

       Licensed under the Apache License, Version 2.0 (the "License");
       you may not use this file except in compliance with the License.
       You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

       Unless required by applicable law or agreed to in writing, software
       distributed under the License is distributed on an "AS IS" BASIS,
       WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
       See the License for the specific language governing permissions and
       limitations under the License.
*/
using Android.App;
using Android.Content;
using Android.Util;
using Android.Widget;
using Huawei.Hms.Push;

namespace XamarinHmsRemoteConfig
{
    [Service]
    [IntentFilter(new[] { "com.huawei.push.action.MESSAGING_EVENT" })]
    public class RemoteConfigMessageService : HmsMessageService
    {

        private static readonly string TAG = "My Messaging Service";

        public override void OnNewToken(string token)
        {
            base.OnNewToken(token);
            Log.Info(TAG, "receive token :" + token);

        }


        public override void OnMessageReceived(RemoteMessage message)
        {
            Log.Info(TAG, "onMessageReceived is called");
            if (message == null)
            {
                Log.Error(TAG, "Received message entity is null!");
                return;
            }

            Log.Info(TAG, "getCollapseKey: " + message.CollapseKey
                    + "\n getData: " + message.Data
                    + "\n getFrom: " + message.From
                    + "\n getTo: " + message.To
                    + "\n getMessageId: " + message.MessageId
                    + "\n getOriginalUrgency: " + message.OriginalUrgency
                    + "\n getUrgency: " + message.Urgency
                    + "\n getSendTime: " + message.SentTime
                    + "\n getMessageType: " + message.MessageType
                    + "\n getTtl: " + message.Ttl);

            RemoteMessage.Notification notification = message.GetNotification();
            if (notification != null)
            {
                Log.Info(TAG, "\n getImageUrl: " + notification.ImageUrl
                        + "\n getTitle: " + notification.Title
                        + "\n getTitleLocalizationKey: " + notification.TitleLocalizationKey
                        + "\n getBody: " + notification.Body
                        + "\n getBodyLocalizationKey: " + notification.BodyLocalizationKey
                        + "\n getIcon: " + notification.Icon
                        + "\n getSound: " + notification.Sound
                        + "\n getTag: " + notification.Tag
                        + "\n getColor: " + notification.Color
                        + "\n getClickAction: " + notification.ClickAction
                        + "\n getChannelId: " + notification.ChannelId
                        + "\n getLink: " + notification.Link
                        + "\n getNotifyId: " + notification.NotifyId);
            }

            if (message.DataOfMap.ContainsKey("DATA_STATE"))
            {
                Log.Info(TAG, "DATA_STATE Exist");
                ISharedPreferences sharedPreferences = this.GetSharedPreferences("Remote_Config", FileCreationMode.Private);
                ISharedPreferencesEditor editor = sharedPreferences.Edit();
                editor.PutBoolean("DATA_OLD", true).Apply();
                Toast.MakeText(this, "The Configuration will be refreshed", ToastLength.Short).Show();

            }


        }



    }
}